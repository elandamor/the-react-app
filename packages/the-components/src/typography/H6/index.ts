import { styled, StyledSystemProps, typography } from '@app/theme';

/**
 * @render react
 * @name H6 component
 * @description Heading level 6.
 * @example
 *  <H6>Heading</H6>
 */

const H6 = styled.h6<StyledSystemProps>`
  ${typography};
`;

H6.defaultProps = {
  color: 'text.default',
  fontSize: [3, 4],
};

export default H6;
