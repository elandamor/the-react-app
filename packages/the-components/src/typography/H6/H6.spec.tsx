import { render } from '@testing-library/react';
import React from 'react';
import H6 from './index';

describe('H6', () => {
  it('renders without crashing', () => {
    const { container } = render(<H6>Heading</H6>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('h6');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct text', () => {
    const { getByText } = render(<H6>Heading</H6>);
    expect(getByText('Heading')).toBeTruthy();
  });
});
