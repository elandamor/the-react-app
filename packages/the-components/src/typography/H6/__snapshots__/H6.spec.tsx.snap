// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H6 renders without crashing 1`] = `
.emotion-0 {
  font-size: 20px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 24px;
  }
}

<h6
  class="emotion-0"
>
  Heading
</h6>
`;
